### PR TITLE
Add persistence and logout redirect uri

### PR DESCRIPTION
**Pull request by FranCarstens**
This merge includes the following changes:

1. Addition of `postLogoutRedirectUri` as documented on [Single-page application: Sign-in and Sign-out](https://docs.microsoft.com/en-us/azure/active-directory/develop/scenario-spa-sign-in?tabs=javascript2)
2. Addition of `useLocalStorageCache` as documented on [More about SSO with MSAL](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-js-sso)
3. Typo corrections of `Exaple` to `Example` in Readme and example files.
4. Updated Readme to include new properties `postLogoutRedirectUri` and `useLocalStorageCache`.

### CODE REVIEW

File: src/MicrosoftLogin.tsx [hunk 1]
```diff
@@ -114,6 +134,16 @@ const MicrosoftLogin: React.FunctionComponent<MicrosoftLoginProps> = ({
...
   }, []);
 
+  // attempt silent login
+  // return msalInstance to user login handler on reload if token is present
+  useEffect(() => {
+    const clientToken = useLocalStorageCache
+      ? localStorage.getItem("msal.idtoken")
+      : sessionStorage.getItem("msal.idtoken");
+
+    clientToken && getGraphAPITokenAndUser(true);
```
Comment (alexandrtovmach) [inline src/MicrosoftLogin.tsx:144]:
```suggestion
    clientToken && getGraphAPITokenAndUser(forceRedirectStrategy || checkToIE());
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
Looks amazing, just one suggestion to improve. We should keep flexibility, and do not force redirect flow if it's possible to stay on page.

**Comment by alexandrtovmach** [comment]
@FranCarstens Thanks for contribution :muscle: Seems really useful improvement

Could you check requested changes and let me know what do you think?

**Comment by alexandrtovmach** [comment]
@FranCarstens Could you?)

**Comment by FranCarstens** [comment]
That looks great. Sorry for the late reply :)

**Review by alexandrtovmach — APPROVED** [review]

**Comment by alexandrtovmach** [comment]
@FranCarstens thanks for contribution
Updates will be available in 1.15.0

**Comment by alexandrtovmach** [comment]
Done:
https://www.npmjs.com/package/react-microsoft-login/v/1.15.0

**Pull request merged**